pandas/tests/groupby/test_groupby.py

```diff
@@ -1674,3 +1674,42 @@ def test_tuple_correct_keyerror():
                                                           [3, 4]]))
     with tm.assert_raises_regex(KeyError, "(7, 8)"):
         df.groupby((7, 8)).mean()
+
+
+def test_dup_index_names():
+    # dup. index names in groupby operations should be renamed (GH 19029):
+    df = pd.DataFrame({'date': list(pd.date_range('5.1.2018', '5.3.2018')),
+                       'vals': list(range(3))})
+
+    mi = pd.MultiIndex.from_product([[5], [1, 2, 3]], names=['date0', 'date1'])
+    expected = pd.Series(data=list(range(3)), index=mi, name='vals')
+
+    failed = False
+    try:
+        result = df.groupby([df.date.dt.month, df.date.dt.day])['vals'].sum()
```

Comment:
Same comment as before - don't need this try...catch block. For what it's worth though let's not duplicate tests, so either have this check here in its own method and remove from the other test, or just update the other test and leave this. See which one makes more sense